### ISSUE TITLE
'ignorePattern' doesn't work for array methods

### ISSUE THREAD

**Issue by ulrichb** [jonaskello/eslint-plugin-functional#56]
The option `{ ignorePattern: ["^mutable"] }` doesn't suppress ...

```ts
const mutableArr: string[] = [];

mutableArr.push(""); // => functional/immutable-data
```
since 0.5.2 (and worked in 0.5.1).

... same issue with `mutableArr.splice(...)`

_Originally posted by @ulrichb in https://github.com/jonaskello/eslint-plugin-functional/issues/45#issuecomment-516659928_

**Comment by ulrichb** [comment]
Many thanks for the fix!

**Comment by ulrichb** [comment]
Version 0.5.3 works great!

What do you think about this sample:


```ts
const mutableArray: string[] = [];   // no error
const mutableMultiMap = new Map<string, string[]>(); // => functional/prefer-readonly-type "Only readonly arrays allowed"
```

Should the `ignorePattern: ["^mutable"]` also suppress the second line (_array_ type as generic argument for `Map`)?

I'm not really sure ...

**Comment by ulrichb** [comment]
... another question/example:

```ts
const mutableResult: { [key: string]: string | undefined } = {};  // => 'prefer-readonly-type' for the indexer
```
